core/src/test/scala/integration/kafka/zk/ZkMigrationIntegrationTest.scala

```diff
@@ -612,7 +613,12 @@ class ZkMigrationIntegrationTest {
       val readyFuture = kraftCluster.controllers().values().asScala.head.controller.waitForReadyBrokers(3)
 
       // Allocate a block of producer IDs while in ZK mode
-      val nextProducerId = sendAllocateProducerIds(zkCluster.asInstanceOf[ZkClusterInstance]).get(30, TimeUnit.SECONDS)
+      var nextProducerId = -1L
+ 
+      TestUtils.retry(60000) {
+        assertDoesNotThrow((() => nextProducerId = sendAllocateProducerIds(zkCluster.asInstanceOf[ZkClusterInstance]).get(20, TimeUnit.SECONDS)): Executable)
+      }
+      assertEquals(0, nextProducerId)
```

Comment:
may I know why do we need this `assertEquals(0, nextProducerId)` here?

Reply:
this checks if the nextProducerId is allocated correctly before moving onto the next steps